tensorflow/contrib/seq2seq/python/ops/beam_search_decoder.py

```diff
@@ -560,20 +599,12 @@ def _maybe_sort_array_beams(self, t, parent_ids, sequence_length):
       return t
     shape = t._element_shape[0]
     # pylint: enable=protected-access
-    batch_size = tensor_util.constant_value(self._batch_size)
-    reshaped_shape = tensor_shape.TensorShape(
-        [batch_size, self._beam_width, None])
-    if (shape.ndims >= 1 and shape[0] == batch_size * self._beam_width
-        or shape.ndims >= 2
-        and shape[0] == batch_size and shape[1] == self._beam_width):
+    _check_static_batch_beam_maybe(
+        shape, tensor_util.constant_value(self._batch_size), self._beam_width)
+    t = t.stack()
```

Comment:
you may want to move the stack inside the control_dependencies.  sometimes people use element_shapes that have an unknown primary dimension, and they use `.concat()` instead of `.stack()`.  if someone's doing that, then this stack will fail because the individual elements have different shapes.  if you move the stack inside the control dependencies, they'll get a much better error message at runtime.

Reply:
How would you check the dynamic element shapes on a non stacked array?

Reply:
sigh; good point.  you emit a warning now, right?  perhaps that's enough for anyone who will run into this.  this really needs a new kernel like TensorArrayAllShapesMatch or something; but that's out of scope for this PR.

Reply:
If the elements shape is defined but the first dimensions is unknown it will go up to this point and fail without warning. I don't know how we can cover this case properly for now.

Reply:
i think i prefer turning it into a warning (use tf_logging.warn) and hope that the user sees it when their session fails to run.  that's probably enough.  having unknown first dimension is more likely than a user using concat on a tensorarray; so we don't need to be too conservative here.